redash/query_runner/python.py

```diff
@@ -187,9 +184,9 @@ def run_query(self, query):
             #       One option is to use ETA with Celery + timeouts on workers
             #       And replacement of worker process every X requests handled.
 
-            exec(code) in restricted_globals, script_locals
+            exec(code) in restricted_globals, self._script_locals
 
-            json_data = json.dumps(self._result)
```

Comment:
I had to do it due to scoping issue and the support for the print statement

On Wed, Jul 22, 2015, 17:37 Arik Fraimovich <notifications@github.com>
wrote:

> In redash/query_runner/python.py
> <https://github.com/EverythingMe/redash/pull/503#discussion_r35219495>:
>
> >
> > -            json_data = json.dumps(self._result)
>
> @erans <https://github.com/erans> FYI.
>
> The problem was here: if you set a different object to result in the
> script, then self._result will no longer reference to it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EverythingMe/redash/pull/503/files#r35219495>.
>